### PR TITLE
Demo

### PR DESCRIPTION
**Pull request by IhostVlad**

### CODE REVIEW

File: examples/todolist/index.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        .reduce((result, key) => {
+            result[key] = req.body[key];
+            return result;
+        }, {});
+    const redirectUrl = req.body.returnUrl || '/';
+
+    command.aggregateId = command.aggregateId || uuid.v4();
+
+    aggregates[req.body.aggregateType](command)
+        .then(() => res.redirect(redirectUrl))
```
Comment (DmitryBogomolov) [inline examples/todolist/index.js]: `res.redirect` is called two times; try the following - 
```
aggregates[key](command)
  .catch(err => console.error(err))
  .then(() => res.redirect(url))
```

File: examples/todolist/index.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/* eslint-disable */
```
Comment (DmitryBogomolov) [inline examples/todolist/index.js]: Why? (as for console commands - disable exact lines)

File: examples/todolist/index.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+app.get('/', function (req, res) {
+    return getCardList()
+        .then(inventoryItems => res.render('index', {
+            items: Object.values(inventoryItems)
+        }));
+});
+
+app.get('/:card', function (req, res) {
+    return getCardDetails()
+        .then(items => items.cards[req.params.card])
```
Comment (DmitryBogomolov) [inline examples/todolist/index.js]: `item.cards[key]` is synchronous - there is no need for two `then`s
```
.then(items => res.render('cardDetails', {
  card: items.cards[req.params.card]
}))
```

File: examples/todolist/index.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import commandHandler from 'resolve-command';
+import query from 'resolve-query';
+
+import todoCardAggregate from './aggregates/TodoCard';
+import todoItemAggregate from './aggregates/TodoItem';
+import cardsReadModel from './read-models/cards';
+import cardDetailsReadModel from './read-models/cardDetails';
+
+const setupMiddlewares = (app) => {
+    app.use( bodyParser.json() );
```
Comment (DmitryBogomolov) [inline examples/todolist/index.js]: Where is it used? (*/:card* is the only candidate and looks like it does not use)

**Comment by DmitryBogomolov** [comment]
What about *"Optimize" query side* entry?

There are two projections now - *cards* and *cardDetails*.
*cards* is fully included into *cardDetails* (`TodoCardCreated` and `TodoCardRemoved` are same).

One query (and projection) should be used used instead of two.

File: examples/todolist/index.js (hunk 4 above)
Comment (VadimKhoroshiltsev) [inline examples/todolist/index.js]: In "/command" to get values from body request

**Comment by MrCheater** [comment]
Rename "read-models" to "projections"

File: examples/todolist/aggregates/TodoItem.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        TodoItemCreated: (state, event) => Immutable({ activated: true, cardId: event.cardId }),
+
+        TodoItemRemoved: state => state.setIn(['activated'], false)
+    },
+
+    commands: {
+        create: (state, args) => {
+            if (!args.cardId) throw new Error('no-cardid');
+            if (!args.name) throw new Error('no-name');
+            if (state && state.activated) throw new Error('already-exist');
```
Comment (ralkov) [inline examples/todolist/aggregates/TodoItem.js]: It must be in reSolve

File: examples/todolist/read-models/cardDetails.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+            state.setIn(
+                ['cards', state.mapTodoToCard[event.aggregateId], 'todoList'],
+                state.cards[state.mapTodoToCard[event.aggregateId]].todoList.without(
+                    event.aggregateId
+                )
+            ),
+
+        TodoItemCheckToggled: (state, event) =>
+            state.setIn(
+                [
```
Comment (ralkov) [inline examples/todolist/read-models/cardDetails.js]: Three variants of formatting is not readable

File: examples/todolist/index.js (hunk 4 above)
Comment (DmitryBogomolov) [inline examples/todolist/index.js]: In */command* *urlencoded* is used. Where is *json* used?

**Pull request merged**